wagtail/contrib/redirects/signal_handlers.py

```diff
@@ -39,10 +39,9 @@ def autocreate_redirects_on_slug_change(
         return None
 
     # Determine sites to create redirects for
-    sites = Site.objects.exclude(root_page=instance).filter(id__in=[
-        option.site_id for option in instance._get_site_root_paths(request=instance)
-        if instance.url_path.startswith(option.root_path)
-    ])
+    sites = Site.objects.filter(id__in=[
+        option.site_id for option in instance._get_relevant_site_root_paths(request=instance)
```

Comment:
`TypeError: Page._get_relevant_site_root_paths() got an unexpected keyword argument 'request'`

Should be `cache_object=instance`, no?

Reply:
Ah yes, of course. Late-night coding is rarely a good idea :) It's good that the CI failed at least... the test coverage is at least good :)